### PR TITLE
Update `--env-file` to sky doc

### PR DESCRIPTION
**Pull request by zpoint**
* Add the parameter `--env-file` to `environment-variables.rst` to complete the doc and example.
* Trailing-whitespace hook automatically fixes some white spaces.

Tested [here](https://github.com/skypilot-org/skypilot/pull/4345#discussion_r1841531611)

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: docs/source/running-jobs/environment-variables.rst [hunk 1]
```diff
@@ -24,9 +24,19 @@ You can specify environment variables to be made available to a task in two ways
 
     envs:
       MYVAR: val
-  
+
 - ``--env`` flag in ``sky launch/exec`` :ref:`CLI <cli>` (takes precedence over the above)
 
+- ``--env-file`` flag in ``sky launch/exec`` :ref:`CLI <cli>` (lowest precedence over the above)
```
[the remark this answers is not in the document]
Comment (zpoint) [reply docs/source/running-jobs/environment-variables.rst]: Good catch!

File: docs/source/running-jobs/environment-variables.rst [hunk 2]
```diff
@@ -16,18 +16,9 @@ User-specified environment variables
...
-
-  .. code-block:: yaml
-
-    envs:
-      MYVAR: val
-
-- ``--env`` flag in ``sky launch/exec`` :ref:`CLI <cli>` (takes precedence over the above)
-
-- ``--env-file`` flag in ``sky launch/exec`` :ref:`CLI <cli>` (lowest precedence over the above)
+- ``--env-file`` flag in ``sky launch/exec`` :ref:`CLI <cli>`, path to a `dotenv` file:
```
Comment (concretevitamin) [inline docs/source/running-jobs/environment-variables.rst:21]:
```suggestion
- ``--env-file`` flag in ``sky launch/exec`` :ref:`CLI <cli>`, which is a path to a `dotenv` file:
```

File: docs/source/running-jobs/environment-variables.rst [hunk 3]
```diff
@@ -37,6 +28,15 @@ You can specify environment variables to be made available to a task in two ways
     WANDB_API_KEY=MY_WANDB_API_KEY
     HF_TOKEN=MY_HF_TOKEN
 
+- ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>`: (takes precedence over the above)
```
Comment (concretevitamin) [inline docs/source/running-jobs/environment-variables.rst:31]:
```suggestion
- ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>` (takes precedence over the above):
```

File: docs/source/running-jobs/environment-variables.rst [hunk 4]
```diff
@@ -16,15 +16,25 @@ User-specified environment variables
...
+
+- ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>`: (takes precedence over the above)
 
   .. code-block:: yaml
 
     envs:
       MYVAR: val
-  
+
 - ``--env`` flag in ``sky launch/exec`` :ref:`CLI <cli>` (takes precedence over the above)
```
Comment (concretevitamin) [inline docs/source/running-jobs/environment-variables.rst:38]: By the way, can you double check in code whether this is the case, e.g., `sky launch --env ENV1 --env-file .env --env ENV2`. Does ordering matter?

**Review by concretevitamin — APPROVED** [review]
Thanks @zpoint, LGTM.

File: docs/source/running-jobs/environment-variables.rst (hunk 4 above)
Comment (zpoint) [reply docs/source/running-jobs/environment-variables.rst:38]: `--env` over `--env-file`

image


And then the merged `env` will overrides the `envs` in `config.yaml`

image


The order doesn't matter

```bash
(sky) ➜  skypilot git:(dev/zeping/env_file_flag_to_doc) cat ~/hello-sky/env_test.yaml                                                                                                                   
# SkyPilot YAML configuration for a cluster with one node
workdir: .

resources:
  cloud: aws
  instance_type: t3.small

num_nodes: 1


envs:
  MYVAR: MYVAR_ENVS_IN_YAML
  WANDB_API_KEY: WANDB_API_KEY_ENVS_IN_YAML
  ENV_TEST: ENV_TEST_ENVS_IN_YAML
setup: |
  echo "Running setup."

run: |
  echo "MYVAR: $MYVAR"
  echo "WANDB_API_KEY: $WANDB_API_KEY"
  echo "ENV_TEST: $ENV_TEST"
(sky) ➜  skypilot git:(dev/zeping/env_file_flag_to_doc) cat ~/hello-sky/env_file.env                                                                                                                    
MYVAR=MYVAR_ENV_FILES
WANDB_API_KEY=WANDB_API_KEY_ENV_FILES
ENV_TEST=ENV_TEST_ENV_FILES
(sky) ➜  skypilot git:(dev/zeping/env_file_flag_to_doc) sky launch -c test_env --env MYVAR=MYVAR_CLI --env-file ~/hello-sky/env_file.env --env WANDB_API_KEY=WANDB_API_KEY_CLI ~/hello-sky/env_test.yaml

(task, pid=1928) MYVAR: MYVAR_CLI
(task, pid=1928) WANDB_API_KEY: WANDB_API_KEY_CLI
(task, pid=1928) ENV_TEST: ENV_TEST_ENV_FILES
```

Comment (zpoint) [reply docs/source/running-jobs/environment-variables.rst:39]: Changed the order in the doc @concretevitamin

**Review by concretevitamin — APPROVED** [review]

**Pull request merged**